proposals/xxxx-package-manager-performance-testing.md

```diff
@@ -0,0 +1,63 @@
+# Package Manager Performance Testing Conventions
+
+* Proposal: [SE-XXXX](xxx-package-manager-performance-testing.md)
+* Author: [Ankit Aggarwal](https://github.com/aciidb0mb3r)
+* Review Manager: TBD
+* Status: Discussion
+
+## Introduction
+
+This is a proposal for adding package manager support for building and running performance tests in release mode.
+
+## Motivation
+
+The package manager supports building and running tests using the `swift test` subtool. The sources are built in debug mode with swift compiler's
+`-enable-testing` flag allowing testable imports in test modules (which gives it internal level access to the imported module). 
+This works well for normal unit tests, however for performance tests we are usually interested in release build test results which takes advantages of compiler 
+optimizations.
+
+We can provide an option to compile and run the tests in release mode with `-enable-testing` flag. However that will disable some of the compiler optimizations, which 
+may be of interest to packages having very performance sensitive code (which critically depends on those optimization).
+
+Since performance tests usually take more time to run than other tests, there is also a requirement to have a way to run performance tests separate from unit tests.
+As a concrete example, package manager's own performance tests are disabled from running on Swift CI.
+
+## Proposed solution
+
+We propose to extend the test modules conventions and consider a directory as "performance test module" which has a suffix "PerformanceTests". The tests in these modules will only run when `--performance` flag is passed to `swift test`.
+
+This command will run performance tests in debug configuration:
+
+```
+$ swift test --performance
+```
+
+And this command will run performance tests in release configuration:
+
+```
+$ swift test --performance -c release
+```
+
+It may be desirable to run performance test in release mode with testing enabled. We propose to add a new boolean property (`enablePerformanceTestability`) in manifest to turn on testability for performance tests in release mode, so packages which want to use the feature in performance tests can do so without passing `-Xswiftc -enable-testing` flags. The default value of this property will be `false`:
```

Comment:
It isn't so  much that it is desirable to run performance tests in release mode with testing enabled, we should describe this in terms of "white box" and "black box".

"black box" tests don't require "testability" be enabled, because they should only test public API.

"white box" tests might, because they are allowed to use internal API.

The useful thing to point out is that, for performance tests, we would like to support a mode in which "black box" tests run in the most optimal configuration (with testability disabled).

So, concretely, I would call the manifest flag `requiresTestability` (meaning this module must have testability enabled in order to be able to build), or (I'm less of a fan of this) `isBlackBox: true`.

**NOTE**: This flag applies to non-performance tests, too -- these are the tests which can be built and run in a release build with testability disabled.